sys/include/event/periodic.h

```diff
@@ -36,20 +36,24 @@
 #ifndef EVENT_PERIODIC_H
 #define EVENT_PERIODIC_H
 
+#include <assert.h>
 #include "event.h"
 #include "ztimer/periodic.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+#define EVENT_PERIODIC_FOREVER      UINT32_MAX
```

Comment:
Why not
```suggestion
#define EVENT_PERIODIC_FOREVER      0
```

Reply:
It makes the condition for `ztimer_periodic` to exit a bit weird.

Reply:
> It makes the condition for ztimer_periodic to exit a bit weird.

Because I'm checking on the remaining count to exit. I could count up but same thing.

Reply:
How about

```C
if (event_periodic->count && --event_periodic->count == 0) {
    return !ZTIMER_PERIODIC_KEEP_GOING;
}

return ZTIMER_PERIODIC_KEEP_GOING;
```